e2e-tests/csv-source/delete.hydra

```diff
@@ -0,0 +1,73 @@
+PREFIX api: <https://cube-creator.zazuko.com/vocab#>
+PREFIX hydra: <http://www.w3.org/ns/hydra/core#>
+PREFIX cc: <https://cube-creator.zazuko.com/vocab#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX query: <http://hypermedia.app/query#>
+PREFIX csvw: <http://www.w3.org/ns/csvw#>
+PREFIX schema: <http://schema.org/>
+
+HEADERS {
+  x-user "john-doe"
+  x-permission "pipelines:write"
+}
+
+
+With Class api:EntryPoint {
+    Expect Property cc:projects {
+        With Operation cc:CreateProjectAction {
+            Invoke {
+                Content-Type "text/turtle"
+
+                ```
+                prefix cc: <https://cube-creator.zazuko.com/vocab#>
+                prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+
+                <> cc:projectSourceKind <https://cube-creator.lndo.site/shape/cube-project/create#CSV> ; rdfs:label "Test Project" .
+                ```
+            } => {
+                Expect Status 201
+            }
+        }
+    }
+}
+
+With Class cc:CubeProject {
+    Expect Link cc:csvMapping {
+        Expect Property cc:csvSourceCollection {
+            Expect Operation cc:UploadCSVAction {
+                Invoke {
+                    Content-Type "text/csv"
+                    Content-Disposition "file; filename=test.jpg"
+
+                    ```
+                    A,B,C
+                    1,2,test
+                    1,2,test
+                    1,2,test1
+                    1,2,test2
+                    .
+                    ```
+                } => {
+                    Expect Status 201
+                    
+                }
+              
+            }
+        }
+    }
+}
+
+With Class cc:CSVSource {
+    Expect Operation cc:DeleteCSVSourceAction {
+        Invoke {
+        } => {
+            Expect Status 204
+        }     
+    }
+    Expect Operation cc:GetCSVSourceAction {
+        Invoke {
+        } => {
+            Expect Status 404
```

Comment:
I would expect a 404 but a 405 is returned. Not sure if we should adapt the test or if there is a bug somewhere

Reply:
From a zulip discussion with @tpluscode 
`hydra-box looks up resources+operations by checking a resource named X (that is obviously getting removed) but then it also looks up resources used as objects by other resources. in our case it is a leftover <csv-mapping> cc:csvSource <deleted-source>`